common/taskana-common/src/main/java/pro/taskana/common/api/WorkingTimeSchedule.java

```diff
@@ -0,0 +1,74 @@
+package pro.taskana.common.api;
+
+import java.time.DayOfWeek;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+class WorkingTimeSchedule {
+
+  private final Map<DayOfWeek, SortedSet<LocalTimeInterval>> workingTimeByDayOfWeek =
+      new EnumMap<>(DayOfWeek.class);
+
+  WorkingTimeSchedule(Map<DayOfWeek, Set<LocalTimeInterval>> workingTimeByDayOfWeek) {
+    for (Entry<DayOfWeek, Set<LocalTimeInterval>> dayOfWeekSetEntry :
+        workingTimeByDayOfWeek.entrySet()) {
+      Set<LocalTimeInterval> workSlots = dayOfWeekSetEntry.getValue();
+      for (LocalTimeInterval interval : workSlots) {
+        if (interval != null && !interval.isValid()) {
+          throw new IllegalArgumentException(
+              "The work period doesn't have two LocalTimes for start and end.");
+        }
+      }
```

Comment:
Please make sure that overlapping workSlots are ok.

Reply:
As discussed, Taskana does not allow overlapping work slots. In case such things are configured Taskana raises an exception.